From bd6aa966df0a593b44830582b2e2b1f160768295 Mon Sep 17 00:00:00 2001
From: Diego Pino Garcia <dpino@igalia.com>
Date: Mon, 15 Jan 2024 14:30:51 +0100
Subject: [PATCH] [quartz] Fix: Cairo 1.18.0 doesn't draw italic or bold text
 on Mac

Commit cf351a8a attempted to convert the font generation in
'_cairo_quartz_font_create_for_toy' to use CTFontCreateWithName and that uses
only Postscript Names, meaning with the hyphens.

Commit c6dc5df6 converted back to CGFont. CGFontCreateWithFontName is supposed
to work with Postscript Names, but it seems sometimes it does not.

In case a CGFont cannot be created using Postscript Names, attempt unhyphenated
version of font family name.
---
 src/cairo-quartz-font.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/cairo-quartz-font.c b/src/cairo-quartz-font.c
index 621d3540c..b7efc54de 100644
--- a/src/cairo-quartz-font.c
+++ b/src/cairo-quartz-font.c
@@ -165,6 +165,17 @@ _cairo_quartz_font_face_create_for_toy (cairo_toy_font_face_t   *toy_face,
 	cgFont = CGFontCreateWithFontName (FontName);
 	CFRelease (FontName);
 
+	if (!cgFont) {
+            /* Attempt to create font by replacing hyphens for spaces in font name. */
+            for (size_t i = 0; i < strlen (full_name); i++) {
+                if (full_name[i] == '-')
+                    full_name[i] = ' ';
+            }
+            FontName = CFStringCreateWithCString (NULL, full_name, kCFStringEncodingASCII);
+            cgFont = CGFontCreateWithFontName (FontName);
+            CFRelease (FontName);
+	}
+
 	if (cgFont)
 	    break;
     }
-- 
2.43.0

